### ISSUE TITLE
Issue 1 — Re-run Study 3 as a Controlled ROI-First Ablation

### ISSUE THREAD

**Issue by Kiwi485** [Kiwi485/edge-deid-sec#59]
# ACM RACS 2026 Camera-Ready Revision Plan

## Paper

**Title:** *Toward Edge-Deployable De-identification of Tongue Diagnosis Images: Segmentation Model Selection and Privacy–Utility Proxy Evaluation*

## Goal

This document converts the reviewers' comments into four actionable issues for revising the camera-ready paper.

- **Issue 1–3:** New experiments or evaluations that strengthen the evidence.
- **Issue 4:** Methodology, terminology, claims, figures, references, and writing revisions.
- **Final deliverable:** A revised two-page camera-ready paper.

> Important: Do not finalize the Abstract, Results, Discussion, or Conclusion until the new results from Issues 1–3 are available.

---

# Issue 1 — Re-run Study 3 as a Controlled ROI-First Ablation

## Summary

Re-run Study 3 using a controlled 2×2 training–inference experiment so that the effect of ROI-first preprocessing can be separated from training–inference domain shift.

## Background

Study 3 currently compares two inference pipelines:

```text
Pipeline A — No ROI
Full image
→ resize the full frame to 256×256
→ run the segmentation model
→ generate the de-identified output

Pipeline B — ROI first
Full image
→ locate and crop the mouth/tongue ROI
→ resize the ROI to 256×256
→ run the same segmentation model
→ generate the de-identified output
```

The current manuscript reports approximately:

```text
No-ROI Dice:       0.280
ROI-first Dice:    0.945

No-ROI residual non-tongue ratio:      7.74%
ROI-first residual non-tongue ratio:    0.15%
```

The two inference paths use the same checkpoint, threshold, and postprocessing. However, the checkpoint was trained using ROI-style images in which the tongue occupies a large part of the input.

## Problem

The current No-ROI condition sends a full facial or oral image to a model trained on cropped ROI images. This changes the input distribution:

```text
Training input:
[ tongue occupies most of the image ]

No-ROI test input:
[ face + mouth + tongue + chin + background ]
```

After the full frame is resized to 256×256, the tongue becomes much smaller than it was in the training images. Therefore, the poor No-ROI result may be caused by both:

1. The absence of ROI-first preprocessing.
2. A mismatch between the training and inference domains.

The current comparison cannot separate these two effects. The reported improvement from 0.280 to 0.945 therefore should not be attributed entirely to ROI-first preprocessing.

## Research Question

After controlling the training domain, does ROI-first processing still provide a meaningful benefit in segmentation quality and privacy–utility performance?

## Required Experiment

Conduct the following 2×2 experiment:

| Model training domain | Inference input | Short name | Purpose |
|---|---|---|---|
| ROI images | ROI images | ROI→ROI | Normal ROI-aligned pipeline |
| ROI images | Full-frame images | ROI→Full | Reproduce the original mismatched No-ROI result |
| Full-frame images | Full-frame images | Full→Full | Fair No-ROI baseline |
| Full-frame images | ROI images | Full→ROI | Measure reverse-domain mismatch |

The primary fair comparison is:

```text
ROI→ROI versus Full→Full
```

These two conditions align each model's training domain with its inference domain.

The other two conditions help quantify domain-shift effects:

```text
ROI→Full and Full→ROI
```

## Dataset Preparation

Use the same original image IDs and annotations whenever possible.

### ROI-domain dataset

- Crop the predefined mouth/tongue ROI.
- Transform the corresponding ground-truth mask using the same crop coordinates.
- Resize both the image and mask to 256×256.
- Use the existing ROI-domain split.

### Full-frame-domain dataset

- Use the complete original frame.
- Use the corresponding full-frame tongue mask.
- Resize both the image and mask to 256×256 using the documented interpolation settings.
- Keep the same train, validation, and evaluation image IDs used by the ROI-domain dataset.

### Data-integrity checks

- Confirm that an image from one subject or source is not placed in multiple splits.
- Confirm that image and mask filenames match after preprocessing.
- Inspect several image–mask pairs visually.
- Record the crop algorithm and any fallback rule.
- Save the exact split IDs for reproducibility.

## Controlled Conditions

Except for the image domain, keep the following settings the same:

```yaml
architecture: U-Net + MobileNetV2
input_resolution: 256x256
train_validation_evaluation_ids: identical
random_seed: identical or explicitly recorded
optimizer: identical
learning_rate: identical
batch_size: identical
maximum_epochs: identical
early_stopping_rule: identical
loss_function: identical
data_augmentation: identical where geometrically valid
checkpoint_selection_rule: identical
mask_threshold: identical or selected using the same validation-only procedure
postprocessing: identical
evaluation_code: identical
hardware: identical
software_environment: identical
```

If a setting cannot be identical, record the difference and explain why it is necessary.

## Avoiding Evaluation Leakage

- Do not tune checkpoints using the 15-image evaluation split.
- Do not tune mask thresholds using the 15-image evaluation split.
- Select checkpoints and thresholds using only the training and validation data.
- Treat the 15 images as an **evaluation split**, not a fully untouched final test set, because the same IDs were previously used for architecture comparison.

## Required Metrics

Report at least:

- Dice coefficient
- Intersection over Union (IoU)
- Precision
- Recall
- Non-Tongue Removal Rate
- Residual Non-Tongue Pixel Ratio
- Output Contamination or over-segmentation rate
- Tongue Region Completeness

For image-level metrics, report:

```text
mean ± standard deviation
```

If possible, also report the median and range because the evaluation set is small.

## Result Table Template

| Training | Testing | Dice | IoU | Precision | Recall | Non-Tongue Removal | Residual Non-Tongue | Tongue Completeness |
|---|---|---:|---:|---:|---:|---:|---:|---:|
| ROI | ROI | TBD | TBD | TBD | TBD | TBD | TBD | TBD |
| ROI | Full | TBD | TBD | TBD | TBD | TBD | TBD | TBD |
| Full | Full | TBD | TBD | TBD | TBD | TBD | TBD | TBD |
| Full | ROI | TBD | TBD | TBD | TBD | TBD | TBD | TBD |

## Suggested Qualitative Figure

Select two or three representative images and show:

| Column | Content |
|---|---|
| 1 | Original full-frame image |
| 2 | ROI crop |
| 3 | Ground-truth mask |
| 4 | ROI→ROI prediction |
| 5 | ROI→Full prediction |
| 6 | Full→Full prediction |

Include at least one typical case and one failure case. Do not select only the best results.

### Outcome A — Full→Full remains much worse than ROI→ROI

Example:

```text
ROI→ROI Dice:       0.94
Full→Full Dice:     0.50
```

Interpretation:

> ROI-first processing remains beneficial even after the training and inference domains are aligned. Cropping enlarges the tongue region relative to the fixed 256×256 input and reduces irrelevant facial context.

This outcome supports a stronger ROI-first claim, although the claim should still be limited to the current dataset and configuration.

### Outcome B — Full→Full becomes close to ROI→ROI

Example:

```text
ROI→ROI Dice:       0.94
Full→Full Dice:     0.91
```

Interpretation:

> Much of the original No-ROI degradation was caused by training–inference distribution mismatch rather than the absence of ROI cropping alone.

Do not continue claiming that ROI-first processing independently improved Dice from 0.280 to 0.945.

### Outcome C — Cross-domain conditions are poor, while matched conditions are good

Example:

```text
ROI→ROI:      good
Full→Full:    good
ROI→Full:     poor
Full→ROI:     poor
```

Interpretation:

> Training–inference domain alignment is the dominant factor. ROI cropping may still have operational benefits, but the evidence does not support attributing the entire performance difference to ROI-first processing.

### Outcome D — Full→Full outperforms ROI→ROI

Interpretation:

> The ROI stage may discard useful spatial context or introduce crop errors. The paper should reconsider whether ROI-first processing is necessary and explain the conditions under which it helps or harms performance.

## Paper Sections to Update

Search the manuscript for:

```text
0.280
0.945
7.74%
0.15%
ROI-first
No-ROI
Study 3
domain shift
```

Update at least:

- Abstract
- Study 3 design
- Study 3 results
- Study 3 analysis
- Discussion or cross-study synthesis
- Threats to validity
- Conclusion

## Deliverables

- Full-frame training dataset or reproducible preprocessing script
- Full-frame U-Net + MobileNetV2 checkpoint
- Four-condition prediction outputs
- Per-image metric CSV
- Aggregated result table
- Representative qualitative figure
- Short written interpretation of domain-shift effects
- Updated Study 3 manuscript text

## Definition of Done

- [ ] Create the full-frame training dataset.
- [ ] Verify image–mask alignment.
- [ ] Preserve the same split IDs across domains.
- [ ] Train the full-frame U-Net + MobileNetV2 model.
- [ ] Run ROI-trained → ROI evaluation.
- [ ] Run ROI-trained → full-frame evaluation.
- [ ] Run full-frame-trained → full-frame evaluation.
- [ ] Run full-frame-trained → ROI evaluation.
- [ ] Calculate Dice, IoU, precision, and recall.
- [ ] Calculate the privacy–utility proxy metrics.
- [ ] Report image-level mean ± standard deviation.
- [ ] Analyze matched-domain and cross-domain performance.
- [ ] Update the Abstract.
- [ ] Update Study 3.
- [ ] Update the Discussion.
- [ ] Update the Threats to Validity.
- [ ] Update the Conclusion.

---
